### ISSUE TITLE
`pattern-not-inside` is excluding a match that it should not

### ISSUE THREAD

**Issue by fopinappb** [semgrep/semgrep#5617]
**Describe the bug**

`pattern-not-inside` is excluding a match that it shouldn't.

```
function getChannel() {
    const c = new BroadcastChannel("someToken");
    return c;
}

function broadcastIt() {
    const cc = window;
    // ruleid:x-typescript-postmessage-copy
    cc.postMessage({status: "up"});
}
```

As `cc.postMessage` comes *after* the `getChannel` definition, it is excluded yet `cc = window` not `cc = getChannel()`

**To Reproduce**

https://semgrep.dev/s/029j

**Expected behavior**

All tests to pass which means line 21 should be a match (and it is not)

**What is the priority of the bug to you?**

- [ ] P0: blocking your adoption of Semgrep or workflow
- [x] P1: important to fix or quite annoying
- [ ] P2: regular bug that should get fixed

**Comment by r2c-demo** [comment]
This issue is synced in Linear at https://linear.app/r2c/issue/PA-1560/pattern-not-inside-is-excluding-a-match-that-it-should-not. _Note_: this link is for r2c use only and is not accessible publicly.

**Comment by fopinappb** [comment]
Closing the issue as had a very nice explanation in [r2c slack](https://r2c-community.slack.com/archives/CK86BJ5DW/p1655988737694899) by @LewisArdern  and @IagoAbal 

TL;DR; (for anyone landing here) `...` is greedy match so the block would go up to line 27 and be done

thanks
